src/components/HomePage.jsx

```diff
@@ -1,3 +1,43 @@
+import { useState, useEffect } from "react";
+import { fetchLatestArticles } from "../../utils/api";
+import { useSearchParams } from "react-router-dom";
+
+import LatestArticleCard from "./LatestArticleCard";
+import Header from "./Header";
+
 export default function HomePage() {
-  return <div>in homepage</div>;
+  const [currentArticles, setCurrentArticles] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const order = "asc";
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchLatestArticles(order).then((data) => {
+      setCurrentArticles(data);
+      setIsLoading(false);
+    });
+  }, [order]);
+
+  return (
+    <section className="homepage-container">
+      <Header />
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+        currentArticles.slice(0, 6).map((article) => {
```

Comment:
Nice use of slice here to only display 6 articles on the homepage. This keeps your page looking clean! It's great that you have accounted for this by adding a 'view all articles' button for the user too 😄 The only issue is I can't seem to click on the articles on the homepage. I would change this as this would make for a better user experience.